react-native-purchases-ui/src/index.tsx

```diff
@@ -308,15 +361,96 @@ export default class RevenueCatUI {
+      const subscriptions: { remove: () => void }[] = [];
+
+      if (callbacks.onFeedbackSurveyCompleted) {
+        const subscription = customerCenterEventEmitter.addListener(
+          'onFeedbackSurveyCompleted',
+          (event: { feedbackSurveyOptionId: string }) => callbacks.onFeedbackSurveyCompleted && 
+            callbacks.onFeedbackSurveyCompleted(event)
+        );
+        subscriptions.push(subscription);
+      }
+
+      if (callbacks.onShowingManageSubscriptions) {
+        const subscription = customerCenterEventEmitter.addListener(
+          'onShowingManageSubscriptions',
+          () => callbacks.onShowingManageSubscriptions && callbacks.onShowingManageSubscriptions()
+        );
+        subscriptions.push(subscription);
+      }
+
+      if (callbacks.onRestoreCompleted) {
+        const subscription = customerCenterEventEmitter.addListener(
+          'onRestoreCompleted',
+          (event: { customerInfo: CustomerInfo }) => callbacks.onRestoreCompleted && 
+            callbacks.onRestoreCompleted(event)
+        );
+        subscriptions.push(subscription);
+      }
+
+      if (callbacks.onRestoreFailed) {
+        const subscription = customerCenterEventEmitter.addListener(
+          'onRestoreFailed',
+          (event: { error: PurchasesError }) => callbacks.onRestoreFailed && 
+            callbacks.onRestoreFailed(event)
+        );
+        subscriptions.push(subscription);
+      }
+
+      if (callbacks.onRestoreStarted) {
+        const subscription = customerCenterEventEmitter.addListener(
+          'onRestoreStarted',
+          () => callbacks.onRestoreStarted && callbacks.onRestoreStarted()
+        );
+        subscriptions.push(subscription);
+      }
+
+      if (callbacks.onRefundRequestStarted) {
+        const subscription = customerCenterEventEmitter.addListener(
+          'onRefundRequestStarted',
+          (event: { productIdentifier: string }) => callbacks.onRefundRequestStarted && 
+            callbacks.onRefundRequestStarted(event)
+        );
+        subscriptions.push(subscription);
+      }
+
+      if (callbacks.onRefundRequestCompleted) {
+        const subscription = customerCenterEventEmitter.addListener(
+          'onRefundRequestCompleted',
+          (event: { productIdentifier: string; refundRequestStatus: REFUND_REQUEST_STATUS }) => callbacks.onRefundRequestCompleted && 
+            callbacks.onRefundRequestCompleted(event)
+        );
+        subscriptions.push(subscription);
+      }
+
+      if (callbacks.onManagementOptionSelected) {
+        const subscription = customerCenterEventEmitter.addListener(
+          'onManagementOptionSelected',
+          (event: { option: CustomerCenterManagementOption; url: string }) => callbacks.onManagementOptionSelected && 
+            callbacks.onManagementOptionSelected(event)
+        );
+        subscriptions.push(subscription);
+      }
+
+      // Return a promise that resolves when the customer center is presented
```

Comment:
Hmm does this resolve when it's presented or is dismissed? Considering we're removing the listeners, I would assume the later?

Reply:
yes! when it's dismissed. Good catch!